tests/test_generation_utils.py

```diff
@@ -1095,16 +1095,17 @@ def test_generate_with_head_masking(self):
 
             signature = inspect.signature(model.forward)
             # We want to test only models where encoder/decoder head masking is implemented
-            if set(head_masking.keys()) < set([*signature.parameters.keys()]):
```

Comment:
Hmm, I'm not so sure here to be honest:

`set(head_masking.keys()) > set([*signature.parameters.keys()])` to be means:

"if all forward arguments are present in head_masking keys" which is not correct here IMO as it would always skip this test. I think the correct solution is:

```
if not (set(head_masking.keys()) < set([*signature.parameters.keys()])):
     ...
```